### PR TITLE
Chain Update Feature [QA-331]

### PR DESCRIPTION
**Pull request by alexbakharew**
https://arangodb.atlassian.net/browse/QA-331

What has been done:

1. All packages will be downloaded or checked for presence before actual test launch - we will avoid errors that we have no required package in the middle of testing after few hours.
2. Test for primary version will be executed. After that upgrade_test will be performed.
3. Chain Update - now we can use for upgrade testing 3 or more versions! Example - 3.8.8:3.9.5:3.9.6:3.10.2:3.10.3
4.  `basecfg` was completely removed from `runner.py` because of multiple data duplication.
5. Some minor and obvious simplifications.
6. Some code cleanups and improvements.

### CODE REVIEW

File: release_tester/full_download_upgrade_test.py [hunk 1]
```diff
@@ -15,12 +15,11 @@
 from tools.killall import list_all_processes
 from write_result_table import write_table
 
-
 # pylint: disable=too-many-arguments disable=too-many-locals disable=too-many-branches, disable=too-many-statements
 def upgrade_package_test(
     dl_opts: DownloadOptions,
     primary_version: str,
-    primary_dlstage: str,
+    primary_source: str,
```
Comment (dothebart) [inline release_tester/full_download_upgrade_test.py:22]: I don't like source here. its overlapping too much with sourcecode.

**Review by dothebart — APPROVED** [review]
please lint and change that variablename; aside of that LGTM.

**Comment by naushniki** [comment]
@alexbakharew Should we add the same changes to full_download_upgrade.py?

**Comment by dothebart** [comment]
@naushniki you would need other parameter handling in that case, right? migrate the versions parameter to one which accepts the list similar to `full_download_upgrade_test`?

**Review by MDAlFattah — APPROVED** [review]
It's working fine with the UI test.

**Pull request merged**